samza-kv/src/main/scala/org/apache/samza/storage/kv/KeyValueStorageEngineMetrics.scala

```diff
@@ -38,16 +40,18 @@ class KeyValueStorageEngineMetrics(
   val ranges = newCounter("ranges")
   val snapshots = newCounter("snapshots")
 
-  val getNs = newTimer("get-ns")
-  val getAllNs = newTimer("get-all-ns")
-  val putNs = newTimer("put-ns")
-  val putAllNs = newTimer("put-all-ns")
-  val deleteNs = newTimer("delete-ns")
-  val deleteAllNs = newTimer("delete-all-ns")
+  val timerEnabled = new MetricsConfig(config).getMetricsTimerEnabled
+
+  val getNs = if (timerEnabled) newTimer("get-ns") else null
```

Comment:
Instead of returning null values I'd prefer to do this check at call sites and not update timers. Otherwise we can run into NPEs if caller doesn't always check.